core/variant.cpp

```diff
@@ -906,158 +906,273 @@ void Variant::reference(const Variant &p_variant) {
 	if (this == &p_variant)
 		return;
 
-	clear();
+	// Don't try to clear if we're just going to be recreating the object
+	// straight away again.
+	if (p_variant.type != type) {
```

Comment:
Maybe an early return would have been more elegant than indenting 200 loc?
```
if (p_variant.type == type)
	return;
```
Or do I miss something that makes it insufficient here?

Reply:
We don't return if it's not the same type, we use a different codepath. There's an 'else' clause.

Reply:
Oops, nevermind then :D